superset/dataframe.py

```diff
@@ -14,257 +14,23 @@
-            return False
-
-        if looks_daty(db_type_str):
-            return True
-        if np_dtype and np_dtype.name and looks_daty(np_dtype.name):
-            return True
-        return False
-
-    @classmethod
-    def is_dimension(cls, dtype, column_name):
-        if cls.is_id(column_name):
-            return False
-        return dtype.name in ("object", "bool")
-
-    @classmethod
-    def is_id(cls, column_name):
-        return column_name.startswith("id") or column_name.endswith("id")
-
-    @classmethod
-    def agg_func(cls, dtype, column_name):
-        # consider checking for key substring too.
-        if cls.is_id(column_name):
-            return "count_distinct"
-        if (
-            hasattr(dtype, "type")
-            and issubclass(dtype.type, np.generic)
-            and is_numeric(dtype)
-        ):
-            return "sum"
-        return None
-
-    @property
-    def columns(self):
-        """Provides metadata about columns for data visualization.
-
-        :return: dict, with the fields name, type, is_date, is_dim and agg.
-        """
-        if self.df.empty:
-            return None
-
-        columns = []
-        sample_size = min(INFER_COL_TYPES_SAMPLE_SIZE, len(self.df.index))
-        sample = self.df
-        if sample_size:
-            sample = self.df.sample(sample_size)
-        for col in self.df.dtypes.keys():
-            db_type_str = self._type_dict.get(col) or self.db_type(self.df.dtypes[col])
-            column = {
-                "name": col,
-                "agg": self.agg_func(self.df.dtypes[col], col),
-                "type": db_type_str,
-                "is_date": self.is_date(self.df.dtypes[col], db_type_str),
-                "is_dim": self.is_dimension(self.df.dtypes[col], col),
-            }
 
-            if not db_type_str or db_type_str.upper() == "OBJECT":
-                v = sample[col].iloc[0] if not sample[col].empty else None
-                if isinstance(v, str):
-                    column["type"] = "STRING"
-                elif isinstance(v, int):
-                    column["type"] = "INT"
-                elif isinstance(v, float):
-                    column["type"] = "FLOAT"
-                elif isinstance(v, (datetime, date)):
-                    column["type"] = "DATETIME"
-                    column["is_date"] = True
-                    column["is_dim"] = False
-                # check if encoded datetime
-                if (
-                    column["type"] == "STRING"
-                    and self.datetime_conversion_rate(sample[col])
-                    > INFER_COL_TYPES_THRESHOLD
-                ):
-                    column.update({"is_date": True, "is_dim": False, "agg": None})
-            # 'agg' is optional attribute
-            if not column["agg"]:
-                column.pop("agg", None)
-            columns.append(column)
-        return columns
+def df_to_dict(dframe: pd.DataFrame) -> Dict:
```

Comment:
nit: similar to how `pd` generally refers to `Pandas`, `df` commonly refers to `DataFrame` (as in the method name). Also add types to Dict, e.g. `Dict[str, Any]`.

Reply:
Using `dframe` here due to a linter complaining about `df` as an argument.

Reply:
In reference to my comment below about `df_to_dict` being expected to return a `dict`, shouldn't this in fact return a `List[Dict[str, Any]]`? Looks like mypy missed this as `data` wasn't typed below, and is hence implicitly typed `Any`.